### PR TITLE
Makes map vote and roundstart smart about player count.

### PR DESCRIPTION
**Pull request by moonheart08**
No more Saltern with 30 players, or Knight Ship with 50.

**Screenshots**
aaaa

**Changelog**
:cl:
- add: Maps now know what the minimum and maximum number of players for them is.
- add: You can now vote for the Knight Ship map with 8 or fewer players. 
- fix: You can no longer play on Saltern with 50 people.

### CODE REVIEW

**Comment by moonheart08** [comment]
Things of note: A serious number of players don't bother to press Ready Up, so I'm opting to go by online population for "can we play this map".

**Comment by moonheart08** [comment]
Future work: allowing multiple copies of a single map to be chosen, for example a squad of four knightships for 20 players.

**Comment by moonheart08** [comment]
How the actual fuck did those tests fail???

File: Content.Server/Maps/GameMapManager.cs [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        [Dependency] private readonly IChatManager _chatManager = default!;
+
+        private GameMapPrototype _currentMap = default!;
+        private bool _currentMapForced;
+
+        public void Initialize()
+        {
+            _configurationManager.OnValueChanged(CCVars.GameMap, value =>
+            {
+                Logger.Debug("WH");
```
Comment (mirrorcult) [inline Content.Server/Maps/GameMapManager.cs:31]: a

File: Content.Server/Maps/GameMapManager.cs [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+                if (TryLookupMap(value, out var map))
+                    _currentMap = map;
+                else
+                    throw new ArgumentException("Bad map.");
```
Comment (mirrorcult) [inline Content.Server/Maps/GameMapManager.cs:35]: more detailed error would be nice

File: Resources/Prototypes/Maps/game.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  minPlayers: 0
+  maxPlayers: 20
+  fallback: true
+
+- type: gameMap
+  id: packedstation
+  mapName: Packedstation
+  mapPath: Maps/packedstation.yml
+  minPlayers: 15
+  maxPlayers: 0
```
Comment (mirrorcult) [inline Resources/Prototypes/Maps/game.yml:14]: ..doesn't this mean packed can never be selected?

**Review by mirrorcult — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/Maps/game.yml (hunk 3 above)
Comment (mirrorcult) [reply Resources/Prototypes/Maps/game.yml:14]: maxPlayers should default to Int.MaxValue or w/e

**Comment by moonheart08** [comment]
Actually, sec, this needs localized.

**Review by mirrorcult — APPROVED** [review]

**Pull request merged**